e2e_tests/tests/streaming/test_client.py

```diff
@@ -0,0 +1,76 @@
+import uuid
+
+import pytest
+
+from determined.common.api import bindings
+from determined.common.streams import wire
+from determined.common.streams._client import LomondStreamWebSocket, ProjectSpec, Stream, Sync
+from tests import api_utils
+
+
+@pytest.mark.e2e_cpu_streaming
+def test_client_connection() -> None:
+    sess = api_utils.admin_session()
+    ws = LomondStreamWebSocket(sess)
+    stream = Stream(ws)
+    syncId = "sync1"
+    stream.subscribe(sync_id=syncId, projects=ProjectSpec(workspace_id=1))
+    event = next(stream)
+    assert event == Sync(syncId, False)
+
+    event = next(stream)
+    assert event.id == 1
+    assert event.immutable == True
+    event = next(stream)
+    assert event == Sync(syncId, True)
+
+
+@pytest.mark.e2e_cpu_streaming
+def test_client_connection() -> None:
+    sess = api_utils.admin_session()
+    ws = LomondStreamWebSocket(sess)
+    stream = Stream(ws)
+
+    syncId = "sync1"
+    projectName = "streaming_project"
+    newProjectName = "streaming_project_1"
+
+    resp_w = bindings.post_PostWorkspace(
+        sess, body=bindings.v1PostWorkspaceRequest(name=f"workspace_aug_{uuid.uuid4().hex[:8]}")
```

Comment:
how about `codecs.encode(random.randbytes(4), 'hex')` instead?

Using `uuid.uuid4()` to get random bytes is odd.